jf_agent/jf_jira/jira_download.py

```diff
@@ -356,45 +397,17 @@ def _download_some(thread_num, start_at, end_at):
             batch_size = 1000
             try:
                 while start_at < min(end_at, total_num_issues):
-                    try:
-                        api_resp = jira_connection.search_issues(
-                            f'{issue_jql} order by id asc',
-                            fields=['updated'],
-                            startAt=start_at,
-                            maxResults=batch_size,
-                        )
-                    except (JIRAError, KeyError) as e:
-                        if hasattr(e, 'status_code') and e.status_code < 500:
-                            # something wrong with our request; re-raise
-                            raise
-
-                        # We have seen sporadic server-side flakiness here. Sometimes Jira Server (but not
-                        # Jira Cloud as far as we've seen) will return a 200 response with an empty JSON
-                        # object instead of a JSON object with an "issues" key, which results in the
-                        # `search_issues()` function in the Jira library throwing a KeyError.
-                        #
-                        # Sometimes both cloud and server will return a 5xx.
-                        #
-                        # In either case, reduce the maxResults parameter and try again, on the theory that
-                        # a smaller ask will prevent the server from choking.
-                        batch_size = int(batch_size / 2)
-                        if batch_size > 0:
-                            agent_logging.log_and_print_error_or_warning(
-                                logger, logging.WARNING, msg_args=[batch_size], error_code=3012,
-                            )
-                            continue
-                        else:
-                            agent_logging.log_and_print_error_or_warning(
-                                logger, logging.ERROR, error_code=3022,
-                            )
-                            raise
+                    api_resp = get_issues(jira_connection, issue_jql, start_at, batch_size)
 
                     issue_metadata = {
                         int(iss.id): IssueMetadata(iss.key, parser.parse(iss.fields.updated))
                         for iss in api_resp
                     }
                     all_issue_metadata.update(issue_metadata)
-                    start_at += len(issue_metadata)
+                    if len(issue_metadata) == 0:
+                        start_at += 1
```

Comment:
This could probably use a comment, since it's not clear from the line itself what the rationale is for this. Something like "get_issues reduced batch size all the way down to 1, still didn't successfully fetch the single issue, so we're skipping it"

Reply:
good call, added here: https://github.com/Jellyfish-AI/jf_agent/pull/213/commits/9211a053e9540760a017bdc111ce382f6ad91b67